src/client/clientauth.h

```diff
@@ -0,0 +1,55 @@
+/*
+Minetest
+Copyright (C) 2024 SFENCE, <sfence.software@gmail.com>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#pragma once
+
+#include <string>
+#include "network/networkprotocol.h"
+#include "util/basic_macros.h"
+
+class ClientAuth
+{
+public:
+	ClientAuth();
+	ClientAuth(const std::string &player_name, const std::string &password);
+
+	~ClientAuth();
+	DISABLE_CLASS_COPY(ClientAuth);
+
+	void moveFrom(ClientAuth& other);
```

Comment:
this should be a move constructor/operator instead

Reply:
I want to do it as a move constructor first.
But this allows me to move without allocating and copying memory.

Reply:
I have tried it again with the move constructor. but it does not work without operator =. And I want to keep that operator disabled.

If I am missing something, please be more explicit.

Reply:
example:
![grafik](https://github.com/minetest/minetest/assets/1042418/86da4ca2-176b-489f-8d54-45fda947ef34)

Reply:
I have added two methods:
```
ClientAuth(ClientAuth &&other) { *this = std::move(other); }

ClientAuth &ClientAuth::operator=(ClientAuth &&other)
{
	clear();

	m_is_empty = other.m_is_empty;

	m_srp_verifier = other.m_srp_verifier;
	m_srp_salt = other.m_srp_salt;

	m_legacy_auth_data = other.m_legacy_auth_data;
	m_srp_auth_data = other.m_srp_auth_data;

	other.m_legacy_auth_data = nullptr;
	other.m_srp_auth_data = nullptr;
}

m_auth = std::move(m_new_auth);
```

Compiled well, no warnings. I tried to change the password and it fails in `ClientAuth &ClientAuth::operator=(ClientAuth &&other)` ... 

```
(gdb) 
50              m_legacy_auth_data = other.m_legacy_auth_data;
(gdb) 
51              m_srp_auth_data = other.m_srp_auth_data;
(gdb) 
53              other.m_legacy_auth_data = nullptr;
(gdb) step
54              other.m_srp_auth_data = nullptr;
(gdb) 

Thread 1 "minetest" received signal SIGILL, Illegal instruction.
0x0000555555785de7 in ClientAuth::operator= (this=0x5555572152f8, other=...)
    at /src/GNU_sources/minetest/minetest/src/client/clientauth.cpp:54
54              other.m_srp_auth_data = nullptr;
(gdb) print other
$5 = (ClientAuth &&) @0x555557215350: {m_is_empty = false, 
  m_srp_verifier = "e\341\027\222\264ֆ\272.\227\202\031\244\314\356 \000\273\372<\226\030\324<\323A\232\3730\362\\bע\361ܲ\255\022J\000r\243Zk\277\242\333u\237\335\353\373X\232\341J\005\363\351=H\344\344r\256\3535\210\017\345|\\8j\017\027\264\225 \226y J\034Uf\323w\307'˺\327\217G_\27400\v\361w\037\365\233\337\025\221\333\351H\230\310\362\321;g\r\216k@`\203\346_\227\377\270\210\345GA(\343\356V\365\020\017\v\031\373!ዃm\360ϼ\031\3557\265\236;~\3068\220ϡ\315c\333\332!\326\036\215\233V>\355\252\335w\326\002,64\212\270\247\246\017T\a\300Qm'\004\276X\027\2602"..., m_srp_salt = "\225{\256|Ə1*y\303\004d\363\361\020x", 
  m_legacy_auth_data = 0x0, m_srp_auth_data = 0x0}
(gdb) print *this
$6 = {m_is_empty = false, 
  m_srp_verifier = "e\341\027\222\264ֆ\272.\227\202\031\244\314\356 \000\273\372<\226\030\324<\323A\232\3730\362\\bע\361ܲ\255\022J\000r\243Zk\277\242\333u\237\335\353\373X\232\341J\005\363\351=H\344\344r\256\3535\210\017\345|\\8j\017\027\264\225 \226y J\034Uf\323w\307'˺\327\217G_\27400\v\361w\037\365\233\337\025\221\333\351H\230\310\362\321;g\r\216k@`\203\346_\227\377\270\210\345GA(\343\356V\365\020\017\v\031\373!ዃm\360ϼ\031\3557\265\236;~\3068\220ϡ\315c\333\332!\326\036\215\233V>\355\252\335w\326\002,64\212\270\247\246\017T\a\300Qm'\004\276X\027\2602"..., m_srp_salt = "\225{\256|Ə1*y\303\004d\363\361\020x", 
  m_legacy_auth_data = 0x7fffa8007fc0, m_srp_auth_data = 0x7fffa80090c0}
(gdb) 
```

This looks like the best introduction I have found: https://medium.com/hackernoon/one-shot-learning-of-c-r-value-and-move-27e5d6bcec3b

But I still am confused and have no idea what caused this Illegal Instruction call.